src/network/networkd-dhcp-server.c

```diff
@@ -29,33 +29,91 @@ static bool link_dhcp4_server_enabled(Link *link) {
         if (!link->network)
                 return false;
 
-        if (link->network->bond)
-                return false;
-
         if (link->iftype == ARPHRD_CAN)
                 return false;
 
         return link->network->dhcp_server;
 }
 
-static Address* link_find_dhcp_server_address(Link *link) {
+void network_adjust_dhcp_server(Network *network) {
+        assert(network);
+
+        if (!network->dhcp_server)
+                return;
+
+        if (network->bond) {
+                log_warning("%s: DHCPServer= is enabled for bond slave. "
+                            "Disabling DHCPServer= setting in [Network] section.",
```

Comment:
Why is this referring to the setting, instead of the feature itself? Why not just say "Disabling the DHCP server." ? (Also applies to other places where a similar warning is emitted.)